match/lib/match/runner.rb

```diff
@@ -107,16 +120,44 @@ def run(params)
       storage.clear_changes if storage
     end
 
+    # Used when creating a new certificate or profile
+    def prefixed_working_directory(working_directory)
+      if self.storage_mode == "git"
+        return working_directory
+      else
+        # We fall back to "*", which means certificates and profiles
+        # from all teams that use this bucket would be installed. This is not ideal, but
+        # unless the user provides a `team_id`, we can't know which one to use
+        # This only happens if `readonly` is activated, and no `team_id` was provided
+        @_folder_prefix ||= self.currently_used_team_id
+        if @_folder_prefix.nil?
+          # We use a `@_folder_prefix` variable, to keep state between multiple calls of this
+          # method, as the value won't change. This way the warning is only printed once
+          UI.important("Looks like you run `match` in `readonly` mode, and didn't provide a `team_id`. This will still work, however it is recommended to provide a `team_id` in your Appfile or Matchfile")
+          @_folder_prefix = "*"
+        end
+        return File.join(working_directory, @_folder_prefix)
+      end
+    end
+
+    # Be smart about optional values here
+    # Depending on the storage mode, different vlaues are required
+    def update_optional_values_depending_on_storage_type(params)
+      if params[:storage_mode] != "git"
+        params.option_for_key(:git_url).optional = true
+      end
+    end
+
     def fetch_certificate(params: nil, working_directory: nil)
       cert_type = Match.cert_type_sym(params[:type])
 
-      certs = Dir[File.join(working_directory, "certs", cert_type.to_s, "*.cer")]
```

Comment:
we keep using `prefixed_working_directory(working_directory)` here. Maybe we can Dir.chdir to it? Stick it in a local?

Reply:
In my experience we should avoid changing the directory if possible, as it has caused issues in the past. In particular running `Dir.chdir` within another `Dir.chdir` causes issues, which is something we already do in `fastlane/runner.rb`